apps/api/src/services/notification/email_notification.ts

```diff
@@ -183,3 +191,89 @@ export async function sendNotificationInternal(
     },
   );
 }
+
+
+export async function sendNotificationWithCustomDays(
+  team_id: string,
+  notificationType: NotificationType,
+  daysBetweenEmails: number,
+  bypassRecentChecks: boolean = false,
+) {
+  return withAuth(async (
+    team_id: string,
+    notificationType: NotificationType,
+    daysBetweenEmails: number,
+    bypassRecentChecks: boolean,
+  ) => {
+    const now = new Date();
+    const pastDate = new Date(now.getTime() - daysBetweenEmails * 24 * 60 * 60 * 1000);
+
+    const { data: recentNotifications, error: recentNotificationsError } = await supabase_service
+      .from("user_notifications")
+      .select("*")
+      .eq("team_id", team_id)
+      .eq("notification_type", notificationType)
+      .gte("sent_date", pastDate.toISOString());
```

Comment:
Nonononononono. Absolutely not. We are not doing a DB query on every single CC job insert over 2x. This must be moved to Redis.